### DISCUSSION TITLE
Masonry component

### DISCUSSION THREAD

**Discussion by plbstl** [chakra-ui/chakra-ui#5266] [Ideas]
### Description

Masonry is a layout that maintains a list of content blocks with variable width (although most implementations support only consistent width) and height, sort of like a mason fitting stones in a wall.

![ERIK JOHANSSON](https://i.imgur.com/6Lo8oun.jpeg)
https://www.erikjo.com/work
![Masonry wall fitting](https://s1.qwant.com/thumbr/700x0/3/7/2c8c178cf2b34b12af3c26cf2aa2bc3d9003a54e8ea5f799a68ee2501725a5/gray-brick-wall.png?u=https%3A%2F%2Fcss-tricks.com%2Fwp-content%2Fuploads%2F2019%2F01%2Fgray-brick-wall.png&q=0&b=1&p=0&a=0)

### Problem Statement/Justification

It would be nice to have a built-in component that creates a masonry layout without having to fallback to a set of other libraries

### Proposed Solution or API

https://drafts.csswg.org/css-grid-3/#masonry-layout

https://github.com/mui-org/material-ui/tree/ba2fce43a735f6085e68c2d76ab746a098488862/packages/mui-lab/src/Masonry

### Alternatives

https://github.com/eiriklv/react-masonry-component
https://github.com/jaredLunde/masonic
https://github.com/paulcollett/react-masonry-css

### Additional Information

https://github.com/desandro/masonry
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Masonry_Layout

**Comment by primos63** [comment]
My opinion: There isn't a need to put every type of component into Chakra when Chakra provides enough of the blocks to build it. This opens the door for others to contribute via third-party packages and contributes to choice. I'll leave it up to the maintainers of the project to determine whether or not something like this is truly needed.
In the meantime, here's a solution by Vitalii Malashevskyi with no third party library and not much in the way of code. This could be easily implemented in an individual's project.
[Masonry example](https://codesandbox.io/s/chakra-ui-masonry-grid-with-dogs-mjc5s) from CodeSandbox.

**Comment by malashevskyi** [comment]
One more simple example  [repo](https://github.com/malashevskyi/masonry-with-a-few-lines), [demo](https://masonry-with-a-few-lines.vercel.app/) of masonry, without using breakpoints, so more flexible

**Comment by primos63** [comment]
I'm going to move this to Discussions where those who are interested in having this can upvote it.

**Comment by naz3eh** [comment]
I'm planning to build a UI lib on top of Chakra UI for masonry

**Reply by lostmarinero** [reply]
Did you ever?
